samigo/samigo-app/src/java/org/sakaiproject/tool/assessment/ui/bean/delivery/DeliveryBean.java

```diff
@@ -1859,41 +1911,45 @@ public String next_page()
     forGrade = false;
 
     if (this.actionMode == TAKE_ASSESSMENT
-        || this.actionMode == TAKE_ASSESSMENT_VIA_URL)
+            || this.actionMode == TAKE_ASSESSMENT_VIA_URL)
     {
       syncTimeElapsedWithServer();
-	
+
       SubmitToGradingActionListener listener =
-        new SubmitToGradingActionListener();
+              new SubmitToGradingActionListener();
       try {
-    	  listener.processAction(null);
+        listener.processAction(null);
       }
       catch (FinFormatException e) {
-		  log.debug(e.getMessage());
-		  return "takeAssessment";
-	  }
-	  catch (SaLengthException sae) {
-		  log.debug(sae.getMessage());
-		  return "takeAssessment";
-	  }
+        log.debug(e.getMessage());
+        return "takeAssessment";
+      }
+      catch (SaLengthException sae) {
+        log.debug(sae.getMessage());
+        return "takeAssessment";
+      }
     }
```

Comment:
A common solution is to extract the code that is shared between the two methods into a new convenience method that the other two will call. In this way, you avoid code duplication.